ApiBundle/Controller/NodeController.php

```diff
@@ -66,32 +67,48 @@ public function showOrCreateAction(Request $request, $nodeId, $errorNode)
         $currentSiteDefaultLanguage = $currentSiteManager->getCurrentSiteDefaultLanguage();
         $language = $request->get('language', $currentSiteDefaultLanguage);
         $siteId = $currentSiteManager->getCurrentSiteId();
-        $node = $this->findOneNode($nodeId, $language, $siteId, $request->get('version'));
-        if ($node) {
-            $this->denyAccessUnlessGranted($this->getAccessRole($node));
+        $site = $this->get('open_orchestra_model.repository.site')->findOneBySiteId($siteId);
+        $siteGetLanguage = false;
+
+        if (!is_null($site)) {
+            $aliases = $site->getAliases();
+            foreach ($aliases as $alias) {
+                if ($siteGetLanguage = ($alias->getLanguage() == $language)) {
+                    break;
```

Comment:
@itkg-nanne : you may throw the exception here intead at the end of the function, it may make the code clearer
+ with that exception thrown here, you can remove the following test ``if ($siteGetLanguage) {``